OOPBASICSPART1TASK1/OOPBASICSPART1TASK1/src/com/gyk/java/oop/basics/Employee.java

```diff
@@ -0,0 +1,106 @@
+package com.gyk.java.oop.basics;
+
+import java.util.Date;
+
+enum ContractType {

```

Comment:
1. Added public static ContractType fromValue ...
Now I can add ContractType in both ways:
employee.setContract(ContractType.FULL_TIME);
employeeSecond.setContract(ContractType.fromValue("Part-Time"));
2. I don't understand you worries regarding toString. It will be not allowed to enter invalid value for contract, so there will be no problems with resolving enum in toString method.